packages/api/src/hooks/useTradingPlatformInvestorPasswordChange.ts

```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react';
+import useRequest from '../useRequest';
+
+/** A custom hook that change the Trading Platform Investor Password. */
+const useTradingPlatformInvestorPasswordChange = () => {
+    const { data, ...rest } = useRequest('trading_platform_investor_password_change');
+
+    // Add additional information to the the Trading Platform Investor Password Change response.
+    const modified_data = useMemo(() => {
+        if (!data) return undefined;
+
+        return { ...data };
+    }, [data]);
+
+    return {
+        /** The response and the mutation of the Trading Platform Investor Password Change API request */
+        data: modified_data,
+        ...rest,
+    };
+};
+
```

Comment:
whats the point of this hook? all its doing is returning undefined if data is empty and its wrapped in memo. No better than calling `useRequest('trading_platform_investor_password_change') directly`

Reply:
We need to pass the payload from the component directly, as for data undefined the error handling is not implemented in any of the hooks right `deriv/api/hooks` so it need a mutate function and i think we will have some changes in the api as there are 4 different apis for password change, so given that thought it is implemented in this way. There is always room for improvement. IMO

Reply:
Updated the description as per usage (mutate)
@henry-deriv